mongo/src/test/scala/io/sphere/mongo/format/OptionMongoFormatSpec.scala

```diff
@@ -0,0 +1,104 @@
+package io.sphere.mongo.format
+
+import io.sphere.mongo.generic._
+import org.scalatest.{MustMatchers, OptionValues, WordSpec}
+import io.sphere.mongo.MongoUtils._
+import DefaultMongoFormats._
+
+object OptionMongoFormatSpec {
+
+  case class SimpleClass(value1: String, value2: Int)
+
+  object SimpleClass {
+    implicit val mongo: MongoFormat[SimpleClass] = mongoProduct(apply _)
+  }
+
+  case class ComplexClass(
+    name: String,
+    simpleClass: Option[SimpleClass])
+
+  object ComplexClass {
+    implicit val mongo: MongoFormat[ComplexClass] = mongoProduct(apply _)
+  }
+
+}
+
+
+class OptionMongoFormatSpec extends WordSpec with MustMatchers with OptionValues {
+  import OptionMongoFormatSpec._
+
+  "MongoFormat[Option[_]]" should {
+    "handle presence of all fields" in {
+      val dbo = dbObj(
+        "value1" -> "a",
+        "value2" -> 45
+      )
+      val result = MongoFormat[Option[SimpleClass]].fromMongoValue(dbo)
+      result.value.value1 mustEqual "a"
+      result.value.value2 mustEqual 45
+    }
+
+    "handle presence of all fields mixed with ignored fields" in {
+      val dbo = dbObj(
+        "value1" -> "a",
+        "value2" -> 45,
+        "value3" -> "b"
+      )
+      val json =
+        """{
+          |  "value1": "a",
+          |  "value2": 45,
+          |  "value3": "b"
+          |}
+        """.stripMargin
+      val result = MongoFormat[Option[SimpleClass]].fromMongoValue(dbo)
+      result.value.value1 mustEqual "a"
+      result.value.value2 mustEqual 45
+    }
+
+    "handle presence of not all the fields" in {
+      val dbo = dbObj("value1" -> "a")
+      an[Exception] mustBe thrownBy (MongoFormat[Option[SimpleClass]].fromMongoValue(dbo))
+    }
+
+    "handle absence of all fields" in {
+      val dbo = dbObj()
+      val result = MongoFormat[Option[SimpleClass]].fromMongoValue(dbo)
+      result mustEqual None
+    }
+
+    "handle absence of all fields mixed with ignored fields" in {
+      val dbo = dbObj("value3" -> "a")
+      val result = MongoFormat[Option[SimpleClass]].fromMongoValue(dbo)
+      result mustEqual None
+    }
+
+    "do not ignore fields if no one is expected" in {
```

Comment:
the description is not quite clear to me. bc its a map there is no expected 'shape' and thus no keys should be ignored, right?

Reply:
Yes, we can try to find a better description.
You're right: `Map[String, String]` does not have any expected 'shape' (so no expectation on fields), thus no keys should be ignored.
`consider all fields if the data type does not impose any restriction`?

Reply:
https://github.com/sphereio/sphere-scala-libs/pull/75/commits/04d14ce501620390e5dadb9dbc6a5eeb3c45ecbb